commons/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java

```diff
@@ -33,21 +52,73 @@ public String getCoords() {
         }
 
         if (exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE) == null) {
-            Log.d("Image", "Picture has no GPS info");
-            return null;
-        }
-        else {
+            imageCoordsExists = false;
+            Log.d(TAG, "Picture has no GPS info");
+
+            //Check what user's preference is for automatic location detection
+            boolean gpsPrefEnabled = gpsPreferenceEnabled();
+
+            if (gpsPrefEnabled) {
+                //If pref enabled, set up LocationListener to get current location
+                LocationManager locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+                Criteria criteria = new Criteria();
+                String provider = locationManager.getBestProvider(criteria, true);
+
+                myLocationListener = new MyLocationListener();
+                locationManager.requestLocationUpdates(provider, 400, 1, myLocationListener);
```

Comment:
When I was searching for info on how to use that, I came across this StackOverflow thread: http://stackoverflow.com/questions/7979230/how-to-read-location-only-once-with-locationmanager-gps-and-network-provider-a , for which the highest voted answer suggested not using it. What do you think of his reasons?